supervisor.go

```diff
@@ -0,0 +1,176 @@
+}
+
+// CreateOrUpdate submits an ingestion specification to druid Supervisor API with a pre-configured druid client.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#create-or-update-a-supervisor
+func (s *SupervisorService) CreateOrUpdate(spec InputIngestionSpec) (string, error) {
+	r, err := s.client.NewRequest("POST", supervisorPathPrefix, spec)
+	if err != nil {
+		return "", err
+	}
+	var result CreateSupervisorResponse
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return "", err
+	}
+	return result.SupervisorID, nil
+}
+
+func applySupervisorId(input string, supervisorId string) string {
+	return strings.Replace(input, ":supervisorId", supervisorId, 1)
+}
+
+// GetActiveIDs returns array of active supervisor IDs.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-ids
+func (s *SupervisorService) GetActiveIDs() ([]string, error) {
+	return nil, errors.New("method GetActiveIDs not implemented")
+}
+
+// GetAllActiveSupervisors returns array of active supervisor objects.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-objects
+func (s *SupervisorService) GetAllActiveSupervisors() ([]SupervisorStateWithSpec, error) {
+	return nil, errors.New("method GetAllActiveSpecs not implemented")
+}
+
+// GetAllActiveStates returns an array of supervisor states.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-supervisor-states
+func (s *SupervisorService) GetActiveStates() ([]SupervisorState, error) {
+	return nil, errors.New("method GetAllActiveStates not implemented")
+}
+
+// GetSpec calls druid Supervisor Status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-supervisor-specification
+func (s *SupervisorService) GetSpec(supervisorId string) (OutputIngestionSpec, error) {
+	r, err := s.client.NewRequest("GET", applySupervisorId(supervisorSpecPathPrefix, supervisorId), nil)
+	var result OutputIngestionSpec
+	if err != nil {
+		return result, err
+	}
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+// GetStatus calls druid Supervisor service's Get status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-supervisor-status
+func (s *SupervisorService) GetStatus(supervisorId string) (SupervisorStatus, error) {
+	r, err := s.client.NewRequest("GET", applySupervisorId(supervisorStatusPathPrefix, supervisorId), nil)
+	var result SupervisorStatus
+	if err != nil {
+		return result, err
+	}
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+// GetAuditHistory calls druid Supervisor Status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-audit-history-for-a-specific-supervisor
+func (s *SupervisorService) GetAuditHistory(_supervisorId string) (interface{}, error) {
```

Comment:
I think this is the only remaining method which does return `interface{}` instead of concrete type. Thanks for adapting the rest though!

Reply:
Fixed